services/github/github-directory-file-count.tester.js

```diff
@@ -0,0 +1,75 @@
+'use strict'
+
+const t = (module.exports = require('../tester').createServiceTester())
+
+const mockContents = nock => {
+  const contents = [
+    { path: 'a', type: 'dir' },
+    { path: 'b', type: 'dir' },
+    { path: 'c.js', type: 'file' },
+    { path: 'd.js', type: 'file' },
+    { path: 'e.txt', type: 'file' },
+  ]
+  return nock('https://api.github.com')
+    .get('/repos/badges/shields/contents/services')
+    .reply(200, contents)
+}
+
+t.create('directory file count')
+  .get('/badges/shields/services.json')
+  .expectBadge({
+    label: 'files',
+    message: /^[0-9]+$/,
+  })
+
+t.create('directory file count (root)')
+  .get('/badges/shields.json')
+  .expectBadge({
+    label: 'files',
+    message: /^[0-9]+$/,
+  })
+
+t.create('directory file count (directory not found)')
+  .get('/badges/shields/not_existing_directory.json')
+  .expectBadge({
+    label: 'files',
+    message: 'repo or directory not found',
+  })
+
+t.create('directory file count (not a directory)')
+  .get('/badges/shields/package.json.json')
+  .expectBadge({
+    label: 'files',
+    message: 'not a directory',
+  })
+
+t.create('directory file count (dir type)')
+  .get('/badges/shields/services.json?type=dir')
+  .intercept(mockContents)
+  .expectBadge({
+    label: 'files',
+    message: '2',
+  })
+
+t.create('directory file count (file type)')
+  .get('/badges/shields/services.json?type=file')
+  .intercept(mockContents)
+  .expectBadge({
+    label: 'files',
+    message: '3',
+  })
+
+t.create('directory file count (invalid type)')
+  .get('/badges/shields/services.json?type=submodule')
+  .expectBadge({
+    label: 'files',
+    message: 'invalid query parameter: type',
+  })
```

Comment:
We can remove this test, as the param validation infrastructure is already well tested elsewhere

Reply:
My intension is rather to test that unexpected values (`valid('dir', 'file')`) passed to `type` will result in an error than test the infrastructure itself.

Reply:
I understand that, and what I'm saying is that is already well tested elsewhere within the service class and input param validation code. We'd have a _ton_ of duplicative service tests (that are primarily used as integration tests because of the myriad external platforms/services/APIs we call) if we included tests like this for each and every service that used params. 

I definitely understand the instinct to add such a test here with the lens of validating this one class, but in the broader context of Shields, each and every service test has an impact both in terms of code maintenance and execution during our daily test runs.

Reply:
You explanation have broadened my perspective. I see where you're coming from. The service tests comes with a trade offs. Your points as valid. I've removed this test.